aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1371,6 +1417,93 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda(const Tensor& self, bool som
   return std::make_tuple(U_working_copy, S_working_copy, VT_working_copy);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ lu_solve ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
+static void apply_lu_solve(Tensor& b, const Tensor& lu, const Tensor& pivots, int64_t& info) {
+#ifndef USE_MAGMA
+AT_ERROR("lu_solve: MAGMA library not found in "
+         "compilation. Please rebuild with MAGMA.");
+#else
+  auto b_data = b.data<scalar_t>();
+  auto lu_data = lu.data<scalar_t>();
+
+  auto n = lu.size(-2);
+  auto nrhs = b.size(-1);
+
+  int info_tmp = 0;
+  if (b.dim() == 2) {
+    magma_int_t info = 0;
+    Tensor pivots_tmp = pivots.cpu();
+    magmaLuSolve<scalar_t>(n, nrhs, lu_data, n, pivots_tmp.data<magma_int_t>(), b_data, n, &info_tmp);
+    info = info_tmp;
+  } else {
+    auto pivots_data = pivots.data<magma_int_t>();
+
+    auto b_stride = matrixStride(b);
+    auto lu_stride = matrixStride(lu);
+    auto pivots_stride = pivots.size(-1);
+    magma_int_t batch_size = magma_int_cast(batchCount(b), "batchCount");
+
+    magma_int_t** pivots_array;
+    scalar_t** lu_array;
+    scalar_t** b_array;
+
+    ALLOCATE_ARRAY(pivots_array, magma_int_t*, batch_size, b);
```

Comment:
This needs a rebase, ALLOCATE_ARRAY was updated to use 3 args

Reply:
Sure, this should be fixed in the rebase.